Content.Server/Disposal/Unit/EntitySystems/DisposableSystem.cs

```diff
@@ -48,9 +48,25 @@ public override void Initialize()
             _physicsQuery = GetEntityQuery<PhysicsComponent>();
             _xformQuery = GetEntityQuery<TransformComponent>();
 
+            SubscribeLocalEvent<EntParentChangedMessage>(OnParentChanged);
             SubscribeLocalEvent<DisposalHolderComponent, ComponentStartup>(OnComponentStartup);
         }
 
+        private void OnParentChanged(ref EntParentChangedMessage message)
+        {
+            var meta = MetaData(message.Entity);
+            if ((meta.Flags & MetaDataFlags.InContainer) != 0)
+                return;
+
+            var xform = Transform(message.Entity);
+            if (!TryComp<DisposalHolderComponent>(xform.ParentUid, out var holder))
+                return;
+
+            // A container-less entity is having its parent set to a disposable holder
+            // such as when that entity is being teleported into the holder
+            _containerSystem.Insert(message.Entity, holder.Container);
+        }
+
```

Comment:
This global subscription to parent changes be noticeable on a profiler, please do not do this.

Reply:
Sure, I updated it with a slightly different solution that doesn't rely on the global event. Mind taking another look?